src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,645 @@
+                            else
+                            {
+                                documentUpdate = default;
+                                return false; // TODO refactor
+                            }
+
+                            break;
+                        default:
+                            return TryReplaceWithLocalFunction(source, out documentUpdate);
+                    }
+
+                    return TryConvertInInvocation(
+                              source,
+                             (ExpressionSyntax variableIdentifier, ExpressionSyntax expression) => SyntaxFactory.ExpressionStatement(insideLoopExpression(variableIdentifier, expression)),
+                              initializer,
+                              variableNamePrefix,
+                              out documentUpdate);
+                }
+
+                documentUpdate = default;
+                return false;
+            }
+
+            private bool TryConvertInInvocation(
+                QueryExpressionSyntax source,
+                Func<ExpressionSyntax, ExpressionSyntax, StatementSyntax> leafExpressionCreationMethod,
+                ExpressionSyntax initializer,
+                string variableName,
+                out DocumentUpdate documentUpdate)
+            {
+                var parenthesizedExpression = (ParenthesizedExpressionSyntax)source.Parent;
+                if (parenthesizedExpression.Parent.Parent.Kind() == SyntaxKind.InvocationExpression)
+                {
+                    switch (parenthesizedExpression.Parent.Parent.Parent.Kind())
+                    {
+                        case SyntaxKind.EqualsValueClause:
+                        case SyntaxKind.MultiplyAssignmentExpression:
+                        case SyntaxKind.SimpleAssignmentExpression:
+                            if (TryGetVariableName(parenthesizedExpression.Parent.Parent.Parent, out var identifierExpression))
+                            {
+                                if (TryConvertQueryExpression(source, (ExpressionSyntax expression) => leafExpressionCreationMethod(identifierExpression, expression), out var qe1))
+                                {
+                                    if (TryFindParentAssignment(source, out var parentStatement))
+                                    {
+                                        var updatedParentExpressionStatement = parentStatement.ReplaceNode(
+                                            source.Parent.Parent.Parent,
+                                            initializer);
+
+                                        documentUpdate = new DocumentUpdate(parentStatement, new[] { updatedParentExpressionStatement, qe1 });
+                                        return true;
+                                    }
+                                }
+                            }
+
+                            break;
+                        case SyntaxKind.ReturnStatement:
```

Comment:
basically, every case would benefit from a short exmaple showing what you're looking for, and what you want to translate to.